pkg/kdo/cli/component/build.go

```
@@ -78,113 +73,228 @@ func (o *BuildIDPOptions) Run() (err error) {
+		}
+		// Wait for pods to start running so that we can tail the logs
+		po, err := o.Context.Client.WaitAndGetPod(watchOptions, corev1.PodRunning, "Waiting for the build job to run")
+		if err != nil {
+			err = errors.New("The build job failed to run")
+			return err
+		}
+
+		err = o.Context.Client.GetPodLogs(po, os.Stdout)
+		if err != nil {
+			err = errors.New("Failed to get the logs for the build job")
+			return err
+		}
+
+		// TODO-KDO: Set owner references
+		var jobSucceeded bool
+		// Loop and see if the job either succeeded or failed
+		loop := true
+		for loop == true {
+			jobs, err := clientset.BatchV1().Jobs(namespace).List(metav1.ListOptions{})
+			if err != nil {
+				panic(err.Error())
+			}
+			for _, job := range jobs.Items {
+				if strings.Contains(job.Name, buildTaskJobName) {
+					succeeded := job.Status.Succeeded
+					failed := job.Status.Failed
+					if succeeded == 1 {
+						fmt.Printf("The job %s succeeded\n", job.Name)
+						jobSucceeded = true
+						loop = false
+					} else if failed > 0 {
+						fmt.Printf("The job %s failed\n", job.Name)
+						jobSucceeded = false
+						loop = false
+					}
 				}
 			}
 		}
-	}
 
-	if loop == false {
-		// delete the job
-		gracePeriodSeconds := int64(0)
-		deletionPolicy := metav1.DeletePropagationForeground
-		err := clientset.BatchV1().Jobs(namespace).Delete(buildTaskJobName, &metav1.DeleteOptions{
-			PropagationPolicy:  &deletionPolicy,
-			GracePeriodSeconds: &gracePeriodSeconds,
-		})
-		if err != nil {
-			panic(err.Error())
-		} else {
-			fmt.Printf("The job %s has been deleted\n", buildTaskJobName)
+		if loop == false {
+			// delete the job
+			gracePeriodSeconds := int64(0)
+			deletionPolicy := metav1.DeletePropagationForeground
+			err := clientset.BatchV1().Jobs(namespace).Delete(buildTaskJobName, &metav1.DeleteOptions{
+				PropagationPolicy:  &deletionPolicy,
+				GracePeriodSeconds: &gracePeriodSeconds,
+			})
+			if err != nil {
+				panic(err.Error())
```

Can we return the error instead of the panic?